### PR TITLE
Added character length limit of program status

### LINKED ISSUE #5000: Status name overflows the 'default status' banner
**Describe the bug**
Given that there seem to be no length limitation on application status names, very long names overflow the 'application submitted' banner.

**To Reproduce**

1. As a CiviForm admin, go to the Programs tab.
2. Find a draft program (or create one if needed).
3. Click on the three dots menu to the right and select "Manage application statuses."
4. Click on the "Create a new status" button.
5. Give your new status a very long name.
6. Select the "Set as default status" checkbox.
7. Click the "Confirm" button.
8. Click "OK" on the "The default status will be updated to..." popup.
9. See the green banner which says:  "<status name> has been updated to the default status."  Notice that the text overflows the banner.

**Expected behavior**
I expect the application status name to fit within the bounds of the banner.  If we want to allow for longer names (not sure why we would need such long names), we could truncate them within the banner.

**Screenshots**
image

**Desktop (please complete the following information):**

- OS: iOS
- Browser: Chrome
- Version: 113.0.5672.126

**Smartphone (please complete the following information):**

Did not verify bug on mobile.

**Additional context**
As a technical note, I would suggest adding a length limit on the status name input field.  Apart from fixing this particular bug, it would prevent other such bugs from occurring.

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Limiting the status of a program to 35 characters. With the change, issue #5000 will get fixed. Below is a screenshot of status being updated to a 35-character-length string not overflowing the banner.
 
Screenshot 2023-12-20 at 11 07 20 AM

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

### Instructions for manual testing

If instructions are needed for manual testing by reviewers, include them here.

### Issue(s) this completes

Fixes #5000

### CODE REVIEW

**Review by avaleske — APPROVED** [review]
thanks for this quick change!

Q: in the description, by "will get fixed as a side project" do you mean "will get fixed by this PR"?

**Comment by azhangggggg** [comment]
> thanks for this quick change!
> 
> Q: in the description, by "will get fixed as a side project" do you mean "will get fixed by this PR"?

Thanks for the quick review! yes I did.

File: browser-test/src/civiform_admin_program_statuses.test.ts [hunk 1]
```diff
@@ -82,6 +82,18 @@ describe('modify program statuses', () => {
       await dismissModal(page)
     })
 
+    it('fails to create status exceeding 35 character limit', async () => {
```
Comment (melanie-exygy) [inline browser-test/src/civiform_admin_program_statuses.test.ts:85]: praise:  Thanks for testing this!

**Review by melanie-exygy — APPROVED** [review]

File: browser-test/src/civiform_admin_program_statuses.test.ts (hunk 1 above)
Comment (azhangggggg) [reply browser-test/src/civiform_admin_program_statuses.test.ts:85]: ofc!

**Pull request merged**